OpenSim/Simulation/SimbodyEngine/Joint.h

```diff
@@ -267,27 +253,36 @@ OpenSim_DECLARE_ABSTRACT_OBJECT(Joint, ModelComponent);
     */
     virtual void scale(const ScaleSet& aScaleSet);
 
-    /**
-    * ModelComponent display interface.
-    *
-    * This method appends the visuals for the Joint to the list appendToThis.
-    * Base class adds geometry for the two Frames making the joint, any extra 
-    * gometry needed to visualize the Joint would be added by the subclass.
+protected:
+    /** A Coordinate place-holder created by constructCoordinate(). E.g.:  
+    class My2DofJoint::Joint {
+        CoordinateP dof1{ constructCoordinate(Coordinate::MotionType::Rotational) };
+        CoordinateP dof2{ constructCoordinate(Coordinate::MotionType::Translatinal) };
+        ...
+    }
     */
-    void generateDecorations(bool fixed, const ModelDisplayHints& hints, const SimTK::State& state,
-        SimTK::Array_<SimTK::DecorativeGeometry>& appendToThis) const override;
-
+    using CoordinateP = SimTK::ReferencePtr<Coordinate>;
+
+    /** Utility for derived Joints to add Coordinate(s) to reflect its DOFs.
+    Derived Joints must construct as many Coordinates as reflected by the
+    Mobilizer Qs unless those Qs are intended to be hidden. */
+    Coordinate* constructCoordinate(Coordinate::MotionType mt) {
+        Coordinate* coord = new Coordinate();
+        coord->setMotionType(mt);
+        std::stringstream name;
+        name << getName() << "_coord_" << get_CoordinateSet().getSize();
+        coord->setName(name.str());
+        // CoordinateSet takes ownership
+        upd_CoordinateSet().adoptAndAppend(coord);
+        return coord;
```

Comment:
Have you thought through and tested what happens when you copy construct, copy assign, etc. a joint? Are you sure the coordinate set doesn't double in size or something silly like that? It seems like you might lose your custom names for the joints. That is, does the following test pass or fail?

```cpp
auto jointA = new PinJoint(...);
jointA->setName("jointA");
jointA->upd_CoordinateSet()[0].setName("flexion");
PinJoint jointB(*jointA);
assert(jointB.get_CoordinateSet()[0].getName() == "flexion");
```